src/useCapabilitiesSafe.ts

```diff
@@ -0,0 +1,36 @@
+import type { Chain } from 'viem';
+import { useAccount } from 'wagmi';
+import { useCapabilities } from 'wagmi/experimental';
+import type { WalletCapabilities } from './types';
+
+export function useCapabilitiesSafe({
+  chain,
+}: {
+  chain: Chain;
+}): WalletCapabilities {
+  const { connector, isConnected } = useAccount();
+
+  // Metamask doesn't support wallet_getCapabilities
+  const isMetamaskWallet = connector?.id === 'io.metamask';
```

Comment:
This feels like whack-a-mole. Should we just try catch and have an empty object?

Reply:
agreed - edited to return `false` for every field depending on the TanStack `error` field from `useCapabilities`

a console error is now emitted from the wallet extension which was previously suppressed by this code

Reply:
agreed - edited to return `false` for every field depending on the TanStack `error` field from `useCapabilities`

a console error is now emitted from the wallet extension which was previously suppressed by this code

Reply:
this code was from Leo on the Basenames team:

```
It would just throw an error
Not breaking the app, but potentially breaking code execution order
```

I don't think this would be the case since we're handling the error here unless there's application specific logic within the extension that breaks on failed `wallet_getCapabilities` requests but I think that's unlikely